notifier/main.py

```diff
@@ -102,22 +125,35 @@ async def _main() -> None:
                 "footer": f"{num_successful_checks}/{num_checks} checks have succeeded",
             }
 
-        response = await post_message(
-            channel=channel,
-            ts=message_ts,
-            icon_emoji=":package:",
-            attachments=[
-                {
-                    "text": status.message,
-                    "author_name": status.author.login,
-                    "author_icon": status.author.avatar_url,
-                    "footer": f"Commit: <{status.url}|`{status.sha[:7]}`>",
-                    "color": commit_color,  # type: ignore
-                },
-                *attachments,
-                summary,
-            ],
-        )
+        response = None
+
+        # Attempt to post notification. If post fails retry provided number of times.
+        for attempt in range(notify_retries):
+            try:
+                response = await post_message(
+                    channel=channel,
+                    ts=message_ts,
+                    icon_emoji=":package:",
+                    attachments=[
+                        {
+                            "text": status.message,
+                            "author_name": status.author.login,
+                            "author_icon": status.author.avatar_url,
+                            "footer": f"Commit: <{status.url}|`{status.sha[:7]}`>",
+                            "color": commit_color,  # type: ignore
+                        },
+                        *attachments,
+                        summary,
+                    ],
+                    timeout=notify_timeout,
+                )
+                break
+            except TimeoutException:
+                if attempt + 1 >= notify_retries:
+                    logger.exception("Failed to notify slack channel")
+                    raise
+
+                logger.error("Unable to post message to slack channel, retrying...")
```

Comment:
Maybe add a slight delay between each try? `await asyncio.sleep(0.5 * (attempt + 1))` or something like that